executor/join.go

```diff
@@ -156,6 +158,9 @@ func (e *HashJoinExec) Close() error {
 
 // Open implements the Executor Open interface.
 func (e *HashJoinExec) Open(ctx context.Context) error {
+	if err := e.baseExecutor.Open(ctx); err != nil {
```

Comment:
Fix https://github.com/pingcap/tidb/issues/27125 in another way, revert https://github.com/pingcap/tidb/pull/27499